src/libs/relay/conduit_relay_zfp.cpp

```diff
@@ -0,0 +1,183 @@
+//
+// * Neither the name of the LLNS/LLNL nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+//-----------------------------------------------------------------------------
+///
+/// file: conduit_relay_zfp.cpp
+///
+//-----------------------------------------------------------------------------
+
+#include "conduit_relay_zfp.hpp"
+#include "zfpfactory.h"
+
+//-----------------------------------------------------------------------------
+// -- begin conduit:: --
+//-----------------------------------------------------------------------------
+namespace conduit
+{
+
+//-----------------------------------------------------------------------------
+// -- begin conduit::relay --
+//-----------------------------------------------------------------------------
+namespace relay
+{
+
+
+//-----------------------------------------------------------------------------
+// -- begin conduit::relay::io --
+//-----------------------------------------------------------------------------
+namespace io
+{
+
+zfp::array*
+unwrap_zfparray(const Node &node)
+{
+    Node compressed_data = node.fetch_child(ZFP_COMPRESSED_DATA_FIELD_NAME);
+
+    // verify word size is readable with zfp
+    // zfp's bitstream consists of uint words
+    bool is_readable = true;
+    switch(stream_word_bits) {
+        case 64:
+            is_readable = compressed_data.dtype().is_uint64();
+            break;
+
+        case 32:
+            is_readable = compressed_data.dtype().is_uint32();
+            break;
+
+        case 16:
+            is_readable = compressed_data.dtype().is_uint16();
+            break;
+
+        case 8:
+            is_readable = compressed_data.dtype().is_uint8();
+            break;
+
+        default:
+            is_readable = false;
+            break;
+    }
+
+    if(!is_readable) {
+        return NULL;
+    }
+
+    zfp::array::header header[1];
```

Comment:
Forgive me if I'm missing something obvious, by why are you using an array containing a single item here instead of declaring a standalone object (i.e. `zfp::array::header` instead of `zfp::array::header[1]`)?

Reply:
I'll change it to remove the array-part